test_builtins.py

```diff
@@ -64,7 +67,11 @@ def test_builtins(self) -> None:
         self.assertEqual(run(env, '(symbol? 42)'), SBool(False))
         self.assertEqual(run(env, '(vector? [])'), SBool(True))
         self.assertEqual(run(env, '(vector? 42)'), SBool(False))
-        self.assertEqual(run(env, '(function? (lambda () []))'), SBool(True))
+        # Use a copy of the environment, since (lambda) adds a name to
```

Comment:
Why is this necessary?

Reply:
If you `run` two programs that define lambdas with the same environment, you'll get a duplicate lambda name. Copying the env whenever that happens makes it robust against future changes that introduce lambdas to the environment.

Reply:
Ah… I expected the lambda names to be globally unique, to my mind it should be perfectly safe to run multiple times.

Reply:
Yeah, I considered making that the case, but then the test cases would start depending on the order they're run in...